liquidation-monitor/src/agent.js

```diff
@@ -0,0 +1,401 @@
+    return [];
+  };
+}
+
+function provideHandleBlock(data) {
+  return async function handleBlock() {
+    const findings = [];
+    const {
+      comptrollerContract,
+      oneInchContract,
+      accounts,
+      tokens,
+      borrow,
+      supply,
+    } = data;
+
+    // Initialize accounts. New accounts will get updated in the block section
+    data.newAccounts.forEach((newAccount) => { accounts[newAccount.toLowerCase()] = {}; });
+    data.totalNewAccounts += data.newAccounts.length;
+    data.newAccounts = [];
+
+    // Update Balances on zero health accounts
+    const filteredAccounts = [];
+    Object.entries(accounts).forEach(([currentAccount, entry]) => {
+      if (entry.health == null || entry.health === 0) {
+        filteredAccounts.push(currentAccount);
+        // Zero account balances
+        Object.keys(supply).forEach((currentToken) => {
+          supply[currentToken][currentAccount] = null;
+        });
+        Object.keys(borrow).forEach((currentToken) => {
+          borrow[currentToken][currentAccount] = null;
+        });
+      }
+    });
+
+    // Grab the assets in first, and make sure they are initialized
+    const foundTokens = [];
+    await Promise.all(filteredAccounts.map(async (currentAccount) => {
+      const assetsIn = await comptrollerContract.getAssetsIn(currentAccount);
+      accounts[currentAccount].assetsIn = assetsIn.map((asset) => {
+        const address = asset.toLowerCase();
+        foundTokens.push(address);
+        return address;
+      });
+    }));
+    // Use Array to Set to Array to remove duplicates
+    const filteredTokens = Array.from(new Set(foundTokens));
+
+    // Initialize token objects
+    await Promise.all(filteredTokens.map(async (token) => {
+      await verifyToken(data, token);
+    }));
+
+    // Grab token balances from on-chain
+    await Promise.all(filteredAccounts.map(async (currentAccount) => {
+      await Promise.all(accounts[currentAccount].assetsIn.map(async (currentToken) => {
+        const snapshot = await tokens[currentToken].contract.getAccountSnapshot(currentAccount);
+
+        // update the supply balance and token quantity
+        supply[currentToken][currentAccount] = new BigNumber(snapshot[1].toString())
+          .dividedBy(tokens[currentToken].cTokenDecimalsMult);
+
+        // update the borrow balance
+        borrow[currentToken][currentAccount] = new BigNumber(snapshot[2].toString());
+      }));
+    }));
+
+    // Update all token prices via 1inch
+    await Promise.all(Object.entries(tokens).map(async ([currentToken, entry]) => {
+      const price = await oneInchContract.getRateToEth(entry.underlying, 0);
+
+      // 1inch's getRateToEth is scaled to 1e18 but is also affected by the underlying token decimal
+      //   using the calculation: 10^18 / 10^(tokenDecimal) - https://docs.1inch.io/docs/spot-price-aggregator/examples
+      //   Combining the two scaling factors: 10^(18 + 18 - tokenDecimal) => 10^(36 - tokenDecimal)
```

Comment:
This is a bit of a verbose suggestion.  I'm trying to make it abundantly clear what the returned value from the 1inch contract is and the different scalings that must be applied to arrive at the price.

Feel free to simplify the comment and reduce the verbosity, as necessary.
```suggestion
      // 1inch's getRateToEth is scaled to 1e18 but is also affected by the underlying token decimal
      //   getRateToEth = (amount of Token / amount of Ether) * 10^18
      // To account for token decimals, Ether decimals, and the scaling, we need to perform the following
      //   price = getRateToEth * (10^tokenDecimals) * (1 / 10^etherDecimals) * (1 / 10^18)
      // Because the number of Ether decimals is fixed at 18, we can simplify the expression
      //   price = getRateToEth * (10^tokenDecimals / 10^36)
      //     or
      //   price = getRateToEth * 10^(tokenDecimals - 36)
      //     or
      //   price = getRateToEth / (10^(36 - tokenDecimals))
      // Ref: https://docs.1inch.io/docs/spot-price-aggregator/examples
```